prepare-operator-metadata.yml

```diff
@@ -21,20 +21,54 @@
     operator_work_dir: "/home/jenkins/agent/test-operator"
     operator_dir: "/tmp/test"
     work_dir: "{{ lookup('env', 'WORKSPACE') }}"
-    jq_bin_path : "/usr/bin/jq"
+    jq_bin_path: "/usr/bin/jq"
     yq_bin_path: "/usr/local/bin/yq"
     testing_bin_path: "/usr/local/bin"
-    current_channel: '' # Added to avoid a bug with undefined variables
+    # Added to avoid a bug with undefined variables
+    current_channel: ''
 
   tasks:
-    
+
     - name: "Set variables for operator_work_dir and operator_dir"
       set_fact:
-        operator_work_dir: "{{ operator_work_dir | default('/home/jenkins/agent/test-operator')}}"
+        operator_work_dir: "{{ operator_work_dir |
+                               default('/home/jenkins/agent/test-operator')}}"
         operator_dir: "{{ operator_dir | default('/tmp/test') }}"
+        parse_operator_metadata_results: {'result': '', 'msg': ''}
+
+    - block:
+        - name: "Run operator-courier nest to copy operator metadata in nested format to the work dir"
+          shell: "operator-courier nest {{ operator_dir }} {{ operator_work_dir }}"
+
+        - name: "Parse operator metadata needed to run the tests"
+          include_role:
+            name: parse_operator_metadata
+
+      rescue:
+        - name: "Rescue block contains the error messages"
+          debug:
+            msg: "Rescue block has found an error, The following are details of failed task."
+        - name: "FAILED task name in ansible is as follows:"
+          debug:
+            msg: "{{ ansible_failed_task.name }}"
+
+        - name: "Result of failed task"
+          debug:
+            msg: "{{ ansible_failed_result }}"
+
+        - name: "Set failure result"
+          set_fact:
+            parse_operator_metadata_results: {'result': 'fail', 'msg': "{{ ansible_failed_result }}", 'description': 'Failed on operator courier nest command check the directory structure'}
```

Comment:
I am sorry but  reporting `ansible_failed_result` is not human friendly.
The message should be self descriptive and it shouls be clear from the `msg` what is wrong and how to fix that.
I personally have huge problem to interpret failure from ansible without ascii colors which I can see locally
And we cannot have colors in returned JSON

Reply:
@lslebodn I have couple of approaches i can take. 

I can make it a yaml string output to make it more readable without colors. 
would that suffice.

Reply:
You probably did not get a point. Changing format ot yaml doe not help much
e.g.
```
[2020-11-04T19:48:05.616Z] 

[2020-11-04T19:48:05.616Z] TASK [parse_operator_metadata : Set variables for csv_path and current_csv_dir] ***

[2020-11-04T19:48:05.875Z] fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'current_csv_path' is undefined\n\nThe error appears to be in '/home/jenkins/agent/workspace/cvp-redhat-operator-metadata-validation-test/operators/config/ansible/roles/parse_operator_metadata/tasks/main.yml': line 105, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: \"Set variables for csv_path and current_csv_dir\"\n    ^ here\n"}

[2020-11-04T19:48:05.875Z] 

[2020-11-04T19:48:05.875Z] TASK [parse_operator_metadata : Rescue block contains the error messages] ******

[2020-11-04T19:48:05.875Z] ok: [localhost] => {

[2020-11-04T19:48:05.875Z]     "msg": "Rescue block has found an error, The following are details of failed task."

[2020-11-04T19:48:05.875Z] }

[2020-11-04T19:48:05.875Z] 

[2020-11-04T19:48:05.875Z] TASK [parse_operator_metadata : FAILED task name in ansible is as follows:] ****

[2020-11-04T19:48:05.875Z] fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'current_csv_path' is undefined\n\nThe error appears to be in '/home/jenkins/agent/workspace/cvp-redhat-operator-metadata-validation-test/operators/config/ansible/roles/parse_operator_metadata/tasks/main.yml': line 188, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: \"FAILED task name in ansible is as follows:\"\n      ^ here\nThis one looks easy to fix. It seems that there is a value started\nwith a quote, and the YAML parser is expecting to see the line ended\nwith the same kind of quote. For instance:\n\n    when: \"ok\" in result.stdout\n\nCould be written as:\n\n   when: '\"ok\" in result.stdout'\n\nOr equivalently:\n\n   when: \"'ok' in result.stdout\"\n"}

[2020-11-04T19:48:05.875Z] 

[2020-11-04T19:48:05.875Z] TASK [Write parse_operator_metadata results to json file] **********************

[2020-11-04T19:48:06.811Z] changed: [localhost] => {"changed": true, "checksum": "b1c892cd7bd3a4a7de3c2458deacc11da33cf325", "dest": "./parse_operator_metadata_results.json", "gid": 0, "group": "root", "md5sum": "ed55e32fba01934aef354cd2235f975c", "mode": "0644", "owner": "default", "size": 35, "src": "/home/jenkins/agent/.ansible/tmp/ansible-tmp-1604519285.95226-295-14232709282158/source", "state": "file", "uid": 1000280000}

[2020-11-04T19:48:06.811Z] 

[2020-11-04T19:48:06.811Z] TASK [Rescue block contains the error messages] ********************************

[2020-11-04T19:48:06.811Z] ok: [localhost] => {

[2020-11-04T19:48:06.811Z]     "msg": "Rescue block has found an error, The following are details of failed task."

[2020-11-04T19:48:06.811Z] }

[2020-11-04T19:48:06.811Z] 

[2020-11-04T19:48:06.811Z] TASK [FAILED task name in ansible is as follows:] ******************************

[2020-11-04T19:48:06.812Z] fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'current_csv_path' is undefined\n\nThe error appears to be in '/home/jenkins/agent/workspace/cvp-redhat-operator-metadata-validation-test/operators/config/ansible/prepare-operator-metadata.yml': line 51, column 11, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n            msg: \"Rescue block has found an error, The following are details of failed task.\"\n        - name: \"FAILED task name in ansible is as follows:\"\n          ^ here\nThis one looks easy to fix. It seems that there is a value started\nwith a quote, and the YAML parser is expecting to see the line ended\nwith the same kind of quote. For instance:\n\n    when: \"ok\" in result.stdout\n\nCould be written as:\n\n   when: '\"ok\" in result.stdout'\n\nOr equivalently:\n\n   when: \"'ok' in result.stdout\"\n"}
```

The issue is with the content. I am sorry but it is not obvious how can fix following error.
people have to check what the playbook try to do and then figure out how to fix it in metadata
```
The task includes an option with an undefined variable. The error was: 'current_csv_path' is undefined 

The error appears to be in '/home/jenkins/agent/workspace/cvp-redhat-operator-metadata-validation-test/operators/config/ansible/prepare-operator-metadata.yml': line 51, column 11, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

            msg: "Rescue block has found an error, The following are details of failed task."
        - name: "FAILED task name in ansible is as follows:"
          ^ here
This one looks easy to fix. It seems that there is a value started
with a quote, and the YAML parser is expecting to see the line ended
with the same kind of quote. For instance:

    when: "ok" in result.stdout

Could be written as:

   when: '"ok" in result.stdout'

Or equivalently:

   when: "'ok' in result.stdout"
```

Reply:
@lslebodn 
The JSON colors are unavoidable when we are running in ansible.  
if we have to disable them we need to disable them thorough out the run. 
However, I have might have a work around to make the output more readable without JSON colors or having a cluttered output. 

The following is the closest readable output i can get from the errors:
```

TASK [parse_operator_metadata : Fail] ***************************************************************************************************************************************
('The failed ansible_task name is Determine the current clustersourceversion '
 'for the operator channel')
'The failed ansible_task name result'
{'_ansible_facts_cacheable': False,
 '_ansible_no_log': False,
 'ansible_facts': {'current_csv': []},
 'changed': False,
 'failed': True,
 'failed_when_result': True}
fatal: [localhost]: FAILED! => {"changed": false, "msg": {"ansible_facts": {"current_csv": []}, "changed": false, "failed": true, "failed_when_result": true}}

TASK [Write parse_operator_metadata results to json file] *******************************************************************************************************************
changed: [localhost]

PLAY RECAP ******************************************************************************************************************************************************************
localhost                  : ok=15   changed=3    unreachable=0    failed=1    skipped=5    rescued=1    ignored=0   

TASK [parse_operator_metadata : Fail] ***************************************************************************************************************************************
('The failed ansible_task name is Determine the current clustersourceversion '
 'for the operator channel')
'The failed ansible_task name result'
{'_ansible_facts_cacheable': False,
 '_ansible_no_log': False,
 'ansible_facts': {'current_csv': []},
 'changed': False,
 'failed': True,
 'failed_when_result': True}
fatal: [localhost]: FAILED! => {"changed": false, "msg": {"ansible_facts": {"current_csv": []}, "changed": false, "failed": true, "failed_when_result": true}}

TASK [Write parse_operator_metadata results to json file] *******************************************************************************************************************
changed: [localhost]

PLAY RECAP ******************************************************************************************************************************************************************
localhost                  : ok=15   changed=3    unreachable=0    failed=1    skipped=5    rescued=1    ignored=0   

```